samples/ChatSample/Views/Home/Index.cshtml

```diff
@@ -16,16 +18,39 @@
 <script>
 let connection = new signalR.HubConnection(`http://${document.location.host}/chat`, 'formatType=json&format=text');
 
-connection.on('Send', function (message) {
+connection.on('SetUsersOnline', function (usersOnline) {
+    for (var user in usersOnline) {
+        addUserOnline(user, usersOnline[user]);
+    }
+});
+
+connection.on('OnConnected', function (id, userName) {
+    var child = document.createElement('li');
+    child.innerText = 'User ' + userName + ' joined the chat';
+    document.getElementById('messages').appendChild(child);
+
+    addUserOnline(id, userName);
+});
+
+connection.on('OnDisconnected', function (id, userName) {
     var child = document.createElement('li');
-    child.innerText = message;
+    child.innerText = 'User ' + userName + ' left the chat';
+    document.getElementById('messages').appendChild(child);
+
+    document.getElementById(id).outerHTML = '';
+});
+
+connection.on('Send', function (userName, message) {
+    var child = document.createElement('li');
+    child.innerHTML = '<b>' + userName + ':</b> ' + message;
```

Comment:
I know this is just a sample but using `innerHTML` for this is dangerous. A user could send arbitrary html and this would inject it into all clients.